commit c18fa2893211d4706911d90c8c64c291f08f33e5
Author: Michael S. Tsirkin <mst@redhat.com>
Date:   Sun Nov 29 17:52:00 2009 +0200

    virtio: set pci bus master enable bit
    
    commit bc505f373979692d51a86d40925f77a8b09d17b9 upstream.
    
    As all virtio devices perform DMA, we
    must enable bus mastering for them to be
    spec compliant.
    
    This patch fixes hotplug of virtio devices
    with Linux guests and qemu 0.11-0.12.
    
    Tested-by: Alexander Graf <agraf@suse.de>
    Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
    Cc: maximilian attems <max@stro.at>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

diff --git a/drivers/virtio/virtio_pci.c b/drivers/virtio/virtio_pci.c
index c7dc37c..19150e7 100644
--- a/drivers/virtio/virtio_pci.c
+++ b/drivers/virtio/virtio_pci.c
@@ -347,6 +347,7 @@ static int __devinit virtio_pci_probe(struct pci_dev *pci_dev,
 		goto out_req_regions;
 
 	pci_set_drvdata(pci_dev, vp_dev);
+	pci_set_master(pci_dev);
 
 	/* we use the subsystem vendor/device id as the virtio vendor/device
 	 * id.  this allows us to use the same PCI vendor/device id for all
